File: src/main/kotlin/org/gameflow/entity/EntityGroups.kt
package org.gameflow.entity

import java.util.Collection
import org.gameflow.pass.Passable

/**
 * A set of named groups of entities.
 */
public trait EntityGroups: Passable {

    public fun groups(): Collection<EntityGroup>
    public fun createGroup(name: String) : EntityGroup
    public fun getGroup(name: String) : EntityGroup
    public fun containsGroup(name: String): Boolean

}

File: src/main/scala/org/gameflow/core/entity/EntityGroups.scala
package org.gameflow.core.entity

import org.gameflow.core.pass.Passable
import org.gameflow.core.LifeCycled

/**
 *
 */
trait EntityGroups extends Passable with LifeCycled {

  def groups(): java.util.Collection[EntityGroup]
  def createGroup(name: Symbol) : EntityGroup
  def getGroup(name: Symbol) : EntityGroup
  def containsGroup(name: Symbol): Boolean
}